lib/spack/docs/packaging_guide.rst

```diff
@@ -3628,18 +3630,51 @@ will likely contain some overriding of default builder methods:
 
    class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
        def cmake_args(self):
-           pass
+           return [
+               self.define_from_variant("MY_FEATURE", "my_feature")
+           ]
 
    class AutotoolsBuilder(spack.build_systems.autotools.AutotoolsBuilder):
        def configure_args(self):
-           pass
+           return self.with_or_without("my-feature", variant="my_feature")
 
-In more complex cases it might happen that the build system changes according to certain conditions,
-for instance across versions. That can be expressed with conditional variant values:
+This way, ``spack install example +feature build_sytem=cmake`` will build using
+``cmake`` and run it with the ``-DMY_FEATURE:BOOL=ON`` define. Alternatively,
+when you run ``spack install example +feature build_system=autotools``, Spack will
+run the configure script with ``./configure --without-my-feature``.
+
+Dependencies are always specified in the package class, because they are usually common to
+all build systems. Build dependencies however can depend on the choice of build system, which
+can be dealt with by using when conditions:
```

Comment:
```suggestion
Dependencies are always specified in the package class. When some dependencies
depend on the choice of the build system, it is possible to use when conditions as 
shown below:
```

(I think the "because" expresses a causality that is not there. The two facts are true, but even if we had completely different dependencies we'd need a single place to get all the "facts" about a package. The distinction was more related to "metadata" and "installation procedure")